pkg/types/stack.go

```diff
@@ -0,0 +1,123 @@
+package types
+
+import (
+	"io/ioutil"
+
+	"gopkg.in/yaml.v2"
+)
+
+type Function struct {
+	// The location of the function handler
+	// relative to context
+	Handler string `yaml:"handler"`
+
+	// TODO(Angus) not in the TypeScript
+	Memory int `yaml:"memory,omitempty"`
+
+	// The build pack version of the membrane used for the function build
+	Version string `yaml:"version,omitempty"`
+
+	// Scripts that will be executed by the nitric
+	// build process before beginning the docker build
+	BuildScripts []string `yaml:"buildScripts,omitempty"`
+
+	// files to exclude from final build
+	Excludes []string `yaml:"excludes,omitempty"`
+
+	// The most requests a single function instance should handle
+	MaxRequests int `yaml:"maxRequests,omitempty"`
+
+	// Simple configuration to determine if the function should be directly
+	// invokable without authentication
+	// would use public, but its reserved by typescript
+	External bool `yaml:"external"`
+}
+
+type Container struct {
+	Dockerfile string   `yaml:"dockerfile"`
+	Args       []string `yaml:"args,omitempty"`
+}
+
+// A subset of a NitricEvent
+// excluding it's requestId
+// This will be generated based on the scedule
+type ScheduleEvent struct {
+	PayloadType string                 `yaml:"payloadType"`
+	Payload     map[string]interface{} `yaml:"payload,omitempty"`
+}
+
+type ScheduleTarget struct {
+	Type string `yaml:"type"` // TODO(Angus) check type: 'topic'; // ; | "queue"
+	Name string `yaml:"name"`
+}
+
+type Schedule struct {
+	Expression string `yaml:"expression"`
+
+	// The Topic to be targeted for schedule
+	Target ScheduleTarget `yaml:"target"`
+	Event  ScheduleEvent  `yaml:"event"`
+	//ext?: Ext; TODO(Angus) check
```

Comment:
The intent behind the `ext` fields in the typescript were generic escape hatches that we could add types for in the provider plugins where necessary:

as an example (would allow the AWS deployment to load a "typesafe" configuration for its own extension fields):
```typescript
type AwsSchedule = Schedule<{'x-aws'?: {
  someUniqueAwsProps: string
}}>
```

This would allow us to add cloud specific escape hatches for finetuning that may not be available in other clouds or are challenging to abstract, without generally impacting deployments to other clouds. (could be argued some such config is crucial and other deployments would be broken without it though :P).

The good news is we don't really have this implemented at the moment, so I think leaving it as you have now is perfectly fine. 👍